### PR TITLE
sanitize templates before displaying them

### LINKED ISSUE #5484: Clicking on a link from a template crashes the client
If I open the templates menu when composing an email, and click on a link in one of the templates, the desktop client crashes. On web, it opens the link in the same browser tab, and the draft is not saved. 

**Steps**:

1. have an account with templates
2. create a template which contains a URL
3. open desktop client > compose new email
4. open templates menu Ctrl + Space
5. search for the template with the link
6. click on link
7. see error

Tested on Linux Desktop client, all recent versions and on web. 

## Test Notes
desktop:
* [ ] when viewing a template in the popup, clicking a link opens the destination in the browser without closing dialogs and without errors
  * this works for autolinks and explicitly linked text
* [ ] when viewing it through the list view, it also does not throw errors and opens the link in the browser

web:
* [ ] when viewing a template in the popup, clicking a link opens the destination in a new browser tab without closing dialogs and without errors
  * this works for autolinks and explicitly linked text
* [ ] when viewing it through the list view, it also does not throw errors and opens the link in a new tab

### PR DESCRIPTION
**Pull request by ganthern**
this also fixes links not opening in the browser or a new tab

fix #5484

### CODE REVIEW

File: src/settings/TemplateDetailsViewer.ts [hunk 1]
```diff
@@ -48,14 +45,14 @@ export class TemplateDetailsViewer implements UpdatableSettingsDetailsViewer {
...
-						value: TEMPLATE_SHORTCUT_PREFIX + neverNull(template.tag),
+						value: TEMPLATE_SHORTCUT_PREFIX + assertNotNull(template.tag, "template without tag!"),
 						disabled: true,
 					}),
 					template.contents.map((emailTemplateContent) => {
 						const language = languageByCode[getLanguageCode(emailTemplateContent)]
 						return m(".flex.flex-column", [
 							m(".h4.mt-l", lang.get(language.textId)),
-							m(".editor-border.text-break", m.trust(emailTemplateContent.text)),
+							m(".editor-border.text-break", m.trust(this.sanitize(emailTemplateContent.text))),
```
Comment (charlag) [inline src/settings/TemplateDetailsViewer.ts:55]: uh oh we should really memoize this

File: src/settings/TemplateDetailsViewer.ts [hunk 2]
```diff
@@ -17,64 +15,66 @@ import { EntityClient } from "../api/common/EntityClient"
...
-							m(".editor-border.text-break", m.trust(emailTemplateContent.text)),
-						])
-					}),
-				]),
-			])
-		}
+	constructor(private readonly template: EmailTemplate, private readonly entityClient: EntityClient, readonly isReadOnly: lazy<boolean>) {
+		this.sanitizedContents = template.contents.map((emailTemplateContent) => ({
+			text: htmlSanitizer.sanitizeHTML(emailTemplateContent.text, { blockExternalContent: false, allowRelativeLinks: true }).html,
+			language: lang.get(languageByCode[getLanguageCode(emailTemplateContent)].textId),
```
Comment (ganthern) [inline src/settings/TemplateDetailsViewer.ts:26]: is storing the translated content here a problem? the view should never survive across the lang switch?

Comment (charlag) [reply src/settings/TemplateDetailsViewer.ts:26]: this is a viewer that is constructed manually so it might actually survive?

Comment (ganthern) [reply src/settings/TemplateDetailsViewer.ts]: that's right, I amended the commit and rebased

**Pull request merged**